### DISCUSSION TITLE
Xtream api proxy - response from player_api.php causing an error in iptv-proxy

### DISCUSSION THREAD

**Discussion by adamsdavis572** [euzu/tuliprox#19] [General]
@euzu i wanted to discuss this here rather than raise an issue as im not sure if this is problem with m3u-filter of not. 

Im progressing implementing m3u-filter with iptv-proxy. it now works well with a URL playlist however im having some issues with Xtream 

when iptv-proxy call the api-proxy for m3u-filter i get an error in iptv proxy from the response from m3u-filter 

```
Error #01: error unmarshaling json: json: invalid use of ,string struct tag, trying to unmarshal unquoted value into xtreamcodes.FlexInt
[GIN] 2023/10/26 - 15:38:15 | 500 |      4.8167ms |    172.29.20.10 | GET      "/player_api.php?username=xxxx&password=xxxx"
```

This seems to be the result of this response 
```
{"user_info":{"active_cons":0,"allowed_output_formats":["ts"],"auth":1,"created_at":1666798274,"exp_date":1729870274,"is_trial":0,"max_connections":1,"message":"Welcome to m3u-filter","password":"xxxx","username":"xxxx","status":"Active"},"server_info":{"url":"192.168.1.199","port":8901,"https_port":443,"server_protocol":"http","rtmp_port":1935,"timezone":"Europe/London","timestamp_now":1698334274,"time_now":"2023-10-26 15:31:14"}}
```

The responser i get from my iptv provider directly is this 

```
{"user_info":{"username":"xxxx","password":"xxxx","message":"","auth":1,"status":"Active","exp_date":"1724658698","is_trial":"0","active_cons":"0","created_at":"1693036298","max_connections":"1","allowed_output_formats":["m3u8","ts"]},"server_info":{"url":"xxxx.xxxx","port":"8080","https_port":"8443","server_protocol":"http","rtmp_port":"8445","timezone":"Europe\/London","timestamp_now":1698335777,"time_now":"2023-10-26 16:56:17"}}
```

the only difference seems to be that some INTs are presented as strings in the response from my IPTV provider. i cant find any doumentation on the types of the response payload for the api call but just wanted to check whether this all looked correct to you? thx

**Comment by euzu** [comment]
I have pushed the changes.  One of the biggest Problems i have is, that i couldnt find a schema desctiption for the xtream api json format.
It took me a while to figure out the field types. I dont know who designed this, but it seems to have a lot of technical debts.

**Comment by adamsdavis572** [comment]
superb - iil test and report back

**Comment by adamsdavis572** [comment]
thanks that call is working now 👍 

however the call to xmltv.php is failing now - is that a call that mp3-filter should handle? 

iptv-proxy 
```
[GIN] 2023/10/26 - 19:39:36 | 500 |     1.98445ms |    172.29.20.10 | GET      "/xmltv.php?username=xxxx&password=xxxx"
Error #01: status code was 404, expected 2XX-3XX
```

m3u-filter
```
[2023-10-26T19:48:15Z INFO  actix_web::middleware::logger] 192.168.1.21 "GET /xmltv.php?username=xxxx&password=xxxx HTTP/1.1" 404 0 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36" 0.000134
[2023-10-26T19:48:15Z DEBUG actix_files::service] error handling /xmltv.php: No such file or directory (os error 2)
```

**Comment by euzu** [comment]
xml..php is on my todo list.
It is currently not handled with n3u-filter.

**Reply by adamsdavis572** [reply]
ok thanks. i really appreciate your efforts and quick responses. your solution is exactly what im looking for and look forward to the new features :-)

**Comment by adamsdavis572** [comment]
one more question... apologies if ive missed the docs for this but im not clear how to build the `/web` directory. when i build with `cargo build` i dont see an output for the web dir. any suggestions ?

**Comment by euzu** [comment]
the frontend is build with nodejs. You need nodejs, then install yarn. Inside the frontemd folder type
```
yarn
yarn build
```
The `build` directory content should be placed under `web` directory.

**Reply by adamsdavis572** [reply]
many thanks

**Comment by euzu** [comment]
[build fe](https://github.com/euzu/m3u-filter/blob/0620aa1fd86578a3e3e661719b8801c51556f11d/bin/build_fe.sh#L2)

there is a Script in the bin folder.
